File: src/main/java/dev/dovydasvenckus/webautomation/config/MigrationConfig.java
package dev.dovydasvenckus.webautomation.config;

import dev.dovydasvenckus.webautomation.AppConfiguration;
import io.dropwizard.db.PooledDataSourceFactory;
import io.dropwizard.migrations.MigrationsBundle;

public class MigrationConfig extends MigrationsBundle<AppConfiguration> {

  @Override
  public PooledDataSourceFactory getDataSourceFactory(AppConfiguration configuration) {
    return configuration.getDataSourceFactory();
  }
}

File: src/main/kotlin/dev/dovydasvenckus/webautomation/config/MigrationConfig.kt
package dev.dovydasvenckus.webautomation.config

import dev.dovydasvenckus.webautomation.AppConfiguration
import io.dropwizard.db.PooledDataSourceFactory
import io.dropwizard.migrations.MigrationsBundle
import java.lang.IllegalArgumentException

class MigrationConfig : MigrationsBundle<AppConfiguration?>() {
    override fun getDataSourceFactory(configuration: AppConfiguration?): PooledDataSourceFactory {
        return configuration?.dataSourceFactory ?: throw IllegalArgumentException("")
    }
}